test/runnable/foreach5.d

```diff
@@ -1085,6 +1087,45 @@ void test14653()
 }
 
 /***************************************/
+// 15777
+
+template funA15777()
+{
+    import imports.test15777a;
+    alias funA15777 = fun;
+}
+
+template funB15777()
+{
+    import imports.test15777b;
+    alias funB15777 = fun;
+}
+
+template funAB15777()
+{
+    import imports.test15777a;
+    import imports.test15777b;
+    alias funAB15777 = fun;
+}
+
+void foo15777(alias tpl)()
+{
+    alias seq = AliasSeq!(tpl!());
+    // Make alias of 'overload set' in tuple elements
```

Comment:
When `tpl` is `funAB15777`, seq.length == 1 and seq[0] is an 'overload set". As you think, it contains two function symbols `imports.tet15777a.fun` and `imports.tet15777b.fun` as the candidates, but the overload set is still one symbol. That's why seq.length is 1.

And, the signature ambiguity of those candidates in an overload set is not shown until the overload set is actually called. The detailed spec about the "overload set" is here: http://dlang.org/spec/function#overload-sets